### PR TITLE
Adicionado uma validação se o content do response estivar vazio(nil),…

### PR DESCRIPTION
**Pull request by slipkbode**
… irá retorno o StatusCode do response como 204 No Content

### CODE REVIEW

File: src/Horse.Jhonson.pas [hunk 1]
```diff
@@ -34,6 +34,10 @@ procedure Jhonson(Req: THorseRequest; Res: THorseResponse; Next: TProc);
   begin
     LWebResponse.Content := TJSONValue(LContent).ToJSON;
     LWebResponse.ContentType := 'application/json';
+  end
+  else if not Assigned(LContent) then
```
Comment (CarlosHe) [inline src/Horse.Jhonson.pas]: I believe that this verification of `Assigned` is not necessary, since the first `if` already does this verification!

Comment (slipkbode) [reply src/Horse.Jhonson.pas]: Bom dia, não me atentei... Obrigado

File: src/Horse.Jhonson.pas [hunk 2]
```diff
@@ -35,7 +35,7 @@ procedure Jhonson(Req: THorseRequest; Res: THorseResponse; Next: TProc);
     LWebResponse.Content := TJSONValue(LContent).ToJSON;
     LWebResponse.ContentType := 'application/json';
   end
-  else if not Assigned(LContent) then
+  else
```
Comment (CarlosHe) [inline src/Horse.Jhonson.pas:38]: There is another observation. If you pass a stream, you will get 204 depending on the order of execution. It is necessary to check if the content is of type "json"...

Comment (slipkbode) [reply src/Horse.Jhonson.pas:38]: Ok... Seria adicionar essa verificação LContent.InheritsFrom(TJSONValue)?

Comment (CarlosHe) [reply src/Horse.Jhonson.pas:38]: It´s can also be... but I believe you can simplify it even more, just checking the type once!

example:

```pascal
    if LContent.InheritsFrom(TJSONValue) then
    begin
      if Assigned(LContent) then
        LWebResponse.Content := TJSONValue(LContent).ToJSON
      else
        LWebResponse.StatusCode := THttpStatus.NoContent.ToInteger;
      LWebResponse.ContentType := 'application/json';
    end;
```

Comment (slipkbode) [reply src/Horse.Jhonson.pas:38]: Nice man, it's great code. Thanks

**Review by snakeice — COMMENTED** [review]
I think this change is not valid when we have a scenario with another middleware, such as stream, jhonson defines 204 and the other already returns something, but it does not define the status code...
Another situation is a POST of a json and the return would be just a 201 to say that the object was created, but I don't want to return anything other than it (hypothetical situation), the jhonson goes there and changes it to 204...

File: src/Horse.Jhonson.pas (hunk 2 above)
Comment (snakeice) [reply src/Horse.Jhonson.pas:38]: If object is nil, application raise exception on verification :/

Comment (slipkbode) [reply src/Horse.Jhonson.pas:38]: Isso eu faria a parte na minha rotina, não teria como implementar para que o horse me devolvesse essa codigo ?

**Comment by viniciussanchez** [comment]
I agree @snakeice 

> I think this change is not valid when we have a scenario with another middleware, such as stream, jhonson defines 204 and the other already returns something, but it does not define the status code...
Another situation is a POST of a json and the return would be just a 201 to say that the object was created, but I don't want to return anything other than it (hypothetical situation), the jhonson goes there and changes it to 204...

**Pull request closed without merging**